### PR TITLE
[enhancement](stmt-kill) cancel query when connection suicide

### PR DESCRIPTION
**Pull request by Hastyshell**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by TangSiyang2001** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 1]
```diff
@@ -1176,20 +1176,14 @@ private void handleKill() throws DdlException {
...
             // Only user itself and user with admin priv can kill connection
-            if (!killCtx.getQualifiedUser().equals(ConnectContext.get().getQualifiedUser())
-                    && !Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(),
-                    PrivPredicate.ADMIN)) {
-                ErrorReport.reportDdlException(ErrorCode.ERR_KILL_DENIED_ERROR, id);
-            }
-
-            killCtx.kill(killStmt.isConnectionKill());
+            ErrorReport.reportDdlException(ErrorCode.ERR_KILL_DENIED_ERROR, id);
         }
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:1185]: why not context.setKilled()?  Is this useless?

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 2]
```diff
@@ -1176,20 +1176,14 @@ private void handleKill() throws DdlException {
         if (killCtx == null) {
             ErrorReport.reportDdlException(ErrorCode.ERR_NO_SUCH_THREAD, id);
         }
-        if (context == killCtx) {
-            // Suicide
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:1180]: in the past, why do we set kill here??? Is this connection killed??

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java (hunk 1 above)
Comment (TangSiyang2001) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:1185]: `killCtx.kill()` contains the `setKill` logic if user wanna kill the conn

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Pull request closed without merging**